keps/sig-node/1592-pod-level-resource-limits/README.md

```diff
@@ -0,0 +1,439 @@
+
+The `npm install` operation was run once before the entire scenario was run.
+
+| Operation     | Response time (sec) w/o pod-level limits | Response time (sec) with pod-level limits | 
+|    :---:      |             :---:                        |        :---:                              |
+| `npm install` |   60  |  43   |
+| `cds watch`   |   5   |  2    |
+| `cds deploy`  |   2   |  0.8  | 
+| `mbt build`   |  90   |  29   |
+
+#### Java
+
+The Java project uses Maven to build and run the project.
+
+A setup operation of `mvn clean install` was performed once, and then the project was either run directly as via `mvn`, or via `cds`.
+
+| Operation           | Response time (sec) w/o pod-level limits | Response time (sec) with pod-level limits | 
+|    :---:            |             :---:                        |            :---:                      |
+| `mvn clean install` | 85 | 42   |
+| `mvn run`           | 50 | 13.5 |
+| `cds deploy`        | 2  | 0.9  |
+
+#### General Load Results
+
+When pod-level resources were defined, the overall CPU utilization (as monitored by Prometheus) was ~70%.
+
+Without pod-level resources, the overall CPU utilization was ~50%.
+
+The Dev-Space load time, measured as the amount of time between Kubernetes launching all of the containers in the Pod and the Dev Space application becoming ready, was also positively affected. Without pod-level resources, the Dev-Space finished loading within 60 seconds, while when pod-level resources were defined, the Dev-Space finished loading in 10 seconds. The explanation for this huge difference is that some of the containers in the pod need to perform one-time startup operations. With pod-level resources defined, these containers can receive CPU resources that are more adequate for the startup operations, while without the pod-level resources, if more CPU is allocated to these containers, then the CPU is not available to the containers intended to run the main workload in the Dev Space. The fact that pod-level resources are available makes it possible to not have to balance load-time vs. general performance for the Dev-Space.
+
+## Implementation History
+
+- 2020-03-04 - v1 of the proposal 
+- 2020-03-06 - Updates due to suggested review
+- 2020-03-15 - More updates due to suggested review
+- 2020-03-17 - Reworked the proposal based on the suggested reviews
+- 2020-03-18 - More updates, after the sig-node meeting on March 17th.
+- 2020-06-04 - Updating the KEP after a trial-run on some real-world workloads
+- 2020-09-22 - Updating the KEP to allow pod-level resource requests
+- 2020-10-07 - Clarified how container-level memory and cpu limits are expected to relate to defined pod limits
+- 2020-11-23 - Updated the KEP to the new KEP format. 
```

Comment:
Is this still targeting 1.31 for alpha?  If so, please copy in the Production Readiness Review Questionnaire section from the template and fill out the alpha sections.

Reply:
@jpbetz https://github.com/kubernetes/enhancements/pull/4678 is the revised version of this KEP that is being worked on actively. May I please add you as the PRR reviewer since John might be busy.

Reply:
Yes, please sign me up